src/tigerbeetle-java/src/main/java/com/tigerbeetle/Accounts.java

```diff
@@ -0,0 +1,317 @@
+     * account.
+     *
+     * @throws IllegalStateException if this batch is read-only.
+     * @throws IndexOutOfBoundsException if exceeds the batch's capacity.
+     */
+    @Override
+    public void add() {
+        super.add();
+    }
+
+    /**
+     * Gets a unique, client-defined identifier for the account.
+     *
+     * @return an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public byte[] getId() {
+        return getUInt128(at(Struct.Id));
+    }
+
+    /**
+     * Gets a unique, client-defined identifier for the account.
+     *
+     * @param part a {@link UInt128} enum indicating which part of the 128-bit value is to be
+     *        retrieved.
+     * @return a {@code long} representing the the first 8 bytes of the 128-bit value if
+     *         {@link UInt128#LeastSignificant} is informed, or the last 8 bytes if
+     *         {@link UInt128#MostSignificant}.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public long getId(final UInt128 part) {
+        return getUInt128(at(Struct.Id), part);
+    }
+
+    /**
+     * Sets a unique, client-defined identifier for the account.
+     *
+     * @param leastSignificant a {@code long} representing the the first 8 bytes of the 128-bit
+     *        value.
+     * @param mostSignificant a {@code long} representing the the last 8 bytes of the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setId(final long leastSignificant, final long mostSignificant) {
+        putUInt128(at(Struct.Id), leastSignificant, mostSignificant);
+    }
+
+    /**
+     * Sets a unique, client-defined identifier for the account.
+     *
+     * @param id an array of 16 bytes representing the 128-bit value.
+     * @throws NullPointerException if {@code id} is null.
+     * @throws IllegalArgumentException if {@code id} is not 16 bytes long.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setId(final byte[] id) {
+
+        if (id == null)
+            throw new NullPointerException("Id cannot be null");
+
+        putUInt128(at(Struct.Id), id);
+    }
+
+
+    /**
+     * Gets an optional secondary identifier to link this account to an external entity.
+     *
+     * @return an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public byte[] getUserData() {
+        return getUInt128(at(Struct.UserData));
+    }
+
+    /**
+     * Gets an optional secondary identifier to link this account to an external entity.
```

Comment:
Same suggestion for these getter/setters, "Gets the optional" rather than "Gets an optional"